### PR TITLE
Fixes progress bar exception preventing managed SDK success messages to show and report.

### LINKED ISSUE #2304: Managed SDK update failure rate increase at or after Nov 22-24
Most of the errors reported are in the format of
```
com.intellij.openapi.diagnostic.RuntimeExceptionWithAttachments: EventQueue.isDispatchThread()\=false Toolkit.getEventQueue()\=com.intellij.ide.IdeEventQueue@526e1f4a Current thread: Thread[pool-3-thread-1\
```

Not containing much useful information.
Need reproduction and evaluation of new error message and a way to extract better information out of `RuntimeExceptionWithAttachments`.

### PR DESCRIPTION
**Pull request by ivanporty**
Fixes #2304.

`dispose()` for progress bar must be called from EDT starting from 2018.3. Tested and works on 2018.2.

### CODE REVIEW

**Comment by ivanporty** [comment]
Tested successfully on 2017.3 as well.

**Comment by etanshaul** [comment]
looks like a possible transient kokoro failure. Kicking off another build manually ..

File: gradle.properties [hunk 1]
```diff
@@ -15,7 +15,7 @@
 #
 
 ideaEdition = IU
-ideaVersion = 2018.2
+ideaVersion = 2018.3
```
Comment (etanshaul) [inline gradle.properties]: @ivanporty was this intentional? I guess it makes sense now that its GA.

Also, I think this is why the build is failing - I need to add the artifacts to our GCS bucket cache.

**Comment by etanshaul** [comment]
looks like there are some test failures in 2018.3 (unrelated to this PR). I think this also explains #2308

**Comment by etanshaul** [comment]
I think I have a fix for this. Will put in a separate PR to fix it

**Comment by etanshaul** [comment]
woohoo. looks like were good now. The EAP build should also be green.

**Comment by etanshaul** [comment]
merging so that I can start the release to test on alpha

**Pull request merged**